lib/ns.c

```diff
@@ -28,24 +28,33 @@
 #include <sched.h>
 #endif
 
+/* for basename */
+#include <libgen.h>
+
 #include "if.h"
 #include "ns.h"
 #include "log.h"
 #include "memory.h"
 
 #include "command.h"
 #include "vty.h"
+#include "vrf.h"
 
-DEFINE_MTYPE_STATIC(LIB, NS, "Logical-Router")
-DEFINE_MTYPE_STATIC(LIB, NS_NAME, "Logical-Router Name")
+DEFINE_MTYPE_STATIC(LIB, NS, "NetNS Context")
+DEFINE_MTYPE_STATIC(LIB, NS_NAME, "NetNS Name")
```

Comment:
Please don't do this. You're using the `ns` structure to represent network namespaces when using the netns VRF backend, but that's not what this structure was meant to be. It was meant to represent logical routers, hence the name above. I'm ok with what you're doing, but you won't make your hack look better by renaming this, just accept the fact that it's a hack.

Reply:
@rwestphal 

Regarding:
> So I was thinking here, what if we create a separate BGP socket for each VRF, even when
> using the VRF-Lite backend?

It could be a good idea, but I believe that it should be done in a separate patch serie in order to avoid confusions. Still, we'd have to be careful to avoid avoid too many sockets opened since it is a resource with some limitations. That's why a separate thread/patch serie seems to be a better option to analyze it properly.

Reply:
@rwestphal @pguibert6WIND 
Regarding:
> Regarding the ioctl calls, most of them are used by the BSDs only, with a few exceptions. But
> even in this case, I think it would be a good idea to create a wrapper function like vrf_ioctl().

I agree that we should wrapp ioctl() to highlight the per VRF behavior. But we shoul not implement the BSD support into this serie to avoid confusions. The wrapper will be a placer holder to starting later a good support for the *BSD.

@pguibert6WIND 
I agree with Renalto that at least the ioctl() should be wrapped (even if void) just ot highlight the needs of further development to be done. What do you think?

Reply:
Hi all,
I will make some changes : 
- split logical router and vrf netns features ( separate files).
- improve the vrf wrapper by ( hiding vrf lite + add vrf_ioctl + hide ns initialisation from vrf)
I kee you in touch.